waveorder/io/reader.py

```diff
@@ -0,0 +1,158 @@
+    def _get_master_ome_tiff(self, folder_):
+        """
+        search all ome.tif in directory
+        check the omexml for element tag corresponding to companion files (not-master-ome)
+
+        :param folder_: full path to folder containing images
+        :return: full path to master-ome tiff
+        """
+
+        from xml.etree import ElementTree as etree  # delayed import
+
+        def tag_search(root_, tag_name='BinaryOnly'):
+            """
+            returns True if tag_name is present
+            """
+            for element in root_:
+                if element.tag.endswith(tag_name):
+                    # todo: question: can we load a full OME-XML from companion file rather than search all files?
+                    print(f'OME series: not an ome-tiff master file')
+                    return True
+            return False
+
+        for file in os.listdir(folder_):
+            print(f"checking {file} for ome-master records")
+            if not file.endswith('.ome.tif'):
+                continue
+            with TiffFile(os.path.join(folder_, file)) as tiff:
+                omexml = tiff.pages[0].description
+                # get omexml root from first page
+                try:
+                    root = etree.fromstring(omexml)
+                except etree.ParseError as exc:
+                    try:
+                        omexml = omexml.decode(errors='ignore').encode()
+                        root = etree.fromstring(omexml)
+                    except Exception as ex:
+                        print(f"Exception while parsing root from omexml: {ex}")
+
+                # search for tag corresponding to non-ome-tiff files
+                if not tag_search(root, "BinaryOnly"):
+                    ome_master = file
+                    break
+                else:
+                    continue
+
+        return os.path.join(folder_, ome_master)
+
+    def get_zarr(self, position=0):
+        if len(self._positions) == 0:
+            self._extract_data(self.master_ome_tiff)
+        # if no position specified, return a full zarr array containing all positions
+        # if position == -1:
+        #     print("no position specified, returning full dataset")
+        #     full_shape = (len(self._positions), ) + self._positions[0].shape
+        #     working_z = zarr.empty(full_shape, chunks=(1, )*(len(full_shape)-2)+full_shape[-2:])
+        #     for pos in range(len(self._positions)):
+        #         working_z[pos] = self._positions[pos]
+        #         return working_z
+        # else:
+        #     return self._positions[position]
+        return self._positions[position]
+
+    def get_array(self, position=0):
+        if len(self._positions) == 0:
+            self._extract_data(self.master_ome_tiff)
+        return np.array(self._positions[position])
+
+    def get_master_ome(self):
+        return self.master_ome_tiff
+
+    def get_num_positions(self):
+        return len(self._positions)
+
```

Comment:
`get_num_positions` will return 0 before using `get_array` or `get_zarr`. Is this something expected?

Reply:
I'm not sure what I intended with this method.  It returns the number of scenes in the multipage-tiff, but as you note, only if you've already initialized with `get_array` or `get_zarr`.  Maybe it's not worth existing?